pipeline.ts

```diff
@@ -1,108 +1,73 @@
-import { BufReader, BufWriter } from "./vendor/https/deno.land/std/io/bufio.ts";
-import {
-  createRequest,
-  readReply,
-  RedisRawReply,
-  ErrorReply,
-} from "./io.ts";
-import { ErrorReplyError } from "./errors.ts";
-import { create } from "./redis.ts";
-import {
-  deferred,
-  Deferred,
-} from "./vendor/https/deno.land/std/async/mod.ts";
-import { RedisCommands } from "./command.ts";
-import { Connection, CommandExecutor } from "./connection.ts";
+import { Connection } from "./connection.ts";
+import { CommandExecutor } from "./executor.ts";
+import { RawReplyOrError, RedisRawReply, sendCommands } from "./io.ts";
+import { Redis, RedisImpl } from "./redis.ts";
+import { Deferred, deferred } from "./vendor/https/deno.land/std/async/mod.ts";
 
-const encoder = new TextEncoder();
-export type RawReplyOrError = RedisRawReply | ErrorReply;
-export type RedisPipeline = {
-  enqueue(command: string, ...args: (number | string)[]): void;
+export type RedisPipeline = Redis & {
   flush(): Promise<RawReplyOrError[]>;
-} & RedisCommands;
+};
 
 export function createRedisPipeline(
-  connection: Connection<RedisRawReply>,
-  opts?: { tx: true },
+  connection: Connection,
+  tx = false,
 ): RedisPipeline {
-  let commands: string[] = [];
-  let queue: {
-    commands: string[];
+  const executor = new PipelineExecutor(connection, tx);
+  async function flush(): Promise<RawReplyOrError[]> {
+    return executor.flush();
+  }
+  const client = new RedisImpl(connection, executor);
+  return Object.assign(client, { flush });
+}
+
```

Comment:
How dou you feel about introducing `Command` interface?
```suggestion
interface Command {
  command: string;
  args: (number | string)[];
  d: Deferred<RedisRawReply>;
}
```